situp-plugins/elasticsearch/README.md

```diff
@@ -20,6 +23,203 @@ pipeline:
+   - `name`: `<index_alias>-index-template`
+   - `index_patterns`: `[<index_alias>]`
+
+### <a name="template_file"></a>Template file (Optional)
+
+A json file path to be read as index template for APM data ingestion. The json file content should be the json value of
+`"template"` key in the json content of elasticsearch [Index templates API](https://www.elastic.co/guide/en/elasticsearch/reference/7.8/index-templates.html).
+
+Default template json (no template file path given) depends on `index_type`:
+
+- `raw`: 
+
+```$xslt
+{
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 2
+  },
+  "mappings": {
+    "date_detection": false,
+    "dynamic_templates": [
+      {
+        "strings_as_keyword": {
+          "mapping": {
+            "ignore_above": 1024,
+            "type": "keyword"
+          },
+          "match_mapping_type": "string"
+        }
+      }
+    ],
+    "_source": {
+      "enabled": true
+    },
+    "properties": {
+      "traceId": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "spanId": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "name": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "kind": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "startTime": {
+        "type": "date_nanos"
+      },
+      "endTime": {
+        "type": "date_nanos"
+      },
+      "resource.attributes.service.name": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      }
+    }
+  }
+}
+```
+
+- `service_map`
+
+```$xslt
+{
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 2
+  },
+  "mappings": {
+    "date_detection": false,
+    "dynamic_templates": [
+      {
+        "strings_as_keyword": {
+          "mapping": {
+            "ignore_above": 1024,
+            "type": "keyword"
+          },
+          "match_mapping_type": "string"
+        }
+      }
+    ],
+    "_source": {
+      "enabled": true
+    },
+    "properties": {
+      "hashId": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "serviceName": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "kind": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      },
+      "destination": {
+        "properties": {
+          "domain": {
+            "ignore_above": 1024,
+            "type": "keyword"
+          },
+          "resource": {
+            "ignore_above": 1024,
+            "type": "keyword"
+          }
+        }
+      },
+      "target": {
+        "properties": {
+          "domain": {
+            "ignore_above": 1024,
+            "type": "keyword"
+          },
+          "resource": {
+            "ignore_above": 1024,
+            "type": "keyword"
+          }
+        }
+      },
+      "traceGroupName": {
+        "ignore_above": 1024,
+        "type": "keyword"
+      }
+    }
+  }
+}
+```
```

Comment:
nit: Can you link to the json file instead, this is not looking good for Readme.

Reply:
Sure. Agreed that should look much better.